### PR TITLE
167 improve file inits and catch operation not found

### PR DESCRIPTION
**Pull request by thomas-senechal**

### CODE REVIEW

File: cli/src/lib/utils/callManager.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    return this.calls.filter((call) => call.status === CallStatus.Error)
+  }
+
+  private async performCall(
+    callUpdate: CallUpdate,
+    onUpdate: (callUpdate: CallUpdate) => void
+  ): Promise<void> {
+    const { call } = callUpdate
+
+    callUpdate.status = CallStatus.Sent
```
Comment (pivilartisant) [inline cli/src/lib/utils/callManager.ts:88]: maybe add a method like setCallStatus() here but not necessary

File: cli/src/lib/utils/callManager.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    onUpdate(callUpdate)
+
+    try {
+      // Execute the call and store the operation
+      const operation = await call.sc.call(
```
Comment (pivilartisant) [inline cli/src/lib/utils/callManager.ts:93]:
```suggestion
      const operation = await operation.sc.call(
```
for a less confusing naming ?

File: cli/src/lib/utils/callManager.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    await Promise.all(activeCalls)
+
+    return this.calls.filter((call) => call.status === CallStatus.Error)
+  }
+
+  private async performCall(
+    callUpdate: CallUpdate,
+    onUpdate: (callUpdate: CallUpdate) => void
+  ): Promise<void> {
+    const { call } = callUpdate
```
Comment (pivilartisant) [inline cli/src/lib/utils/callManager.ts:86]:
```suggestion
    const { call as operation } = callUpdate
```

File: cli/src/lib/utils/callManager.ts (hunk 2 above)
Comment (pivilartisant) [reply cli/src/lib/utils/callManager.ts:93]: or anything else really

**Review by pivilartisant — APPROVED** [review]

File: cli/src/lib/utils/callManager.ts (hunk 2 above)
Comment (thomas-senechal) [reply cli/src/lib/utils/callManager.ts:93]: Clearly needs a better name you are right 🫨

**Pull request merged**